src/main/java/com/xceptance/neodymium/module/statement/browser/BrowserStatement.java

```diff
@@ -285,6 +289,17 @@ private boolean canReuse(boolean preventReuse, WebDriverStateContainer webDriver
             browserAnnotations.addAll(classBrowserAnnotations);
         }
 
+        if (!classRandomBrowserAnnotation.isEmpty() && methodBrowserAnnotations.isEmpty())
+        {
+            Collections.shuffle(browserAnnotations, Neodymium.getRandom());
+            browserAnnotations = browserAnnotations.subList(0, classRandomBrowserAnnotation.get(0).value());
+        }
+        if (!methodRandomBrowserAnnotations.isEmpty())
+        {
+            Collections.shuffle(browserAnnotations, Neodymium.getRandom());
+            browserAnnotations = browserAnnotations.subList(0, methodRandomBrowserAnnotations.get(0).value());
+        }
```

Comment:
Add a check if more browsers then available should be chosen.

Reply:
Done